client/verta/verta/_deployment/endpoint.py

```diff
@@ -76,9 +77,66 @@ def _get_json_by_path(cls, conn, workspace, path):
         # TODO: iterate through response.json().get('endpoints', []) to find matching creator_request.path
 
     def update(self, run, strategy):
-        raise NotImplementedError
-        # TODO: check if isinstance(run, experimentrun.ExperimentRun)
-        # TODO: check if isinstance(strategy, deployment._UpdateStrategy)
+        if not isinstance(run, experimentrun.ExperimentRun):
+            raise TypeError("run must be an ExperimentRun")
+
+        if not issubclass(strategy, deployment._UpdateStrategy):
+            raise TypeError("strategy must be an _UpdateStrategy")
+
+        stage_id = self._get_active_stage()
+
+        # Create new build:
+        url = "{}://{}/api/v1/deployment/workspace/{}/builds".format(
+            self._conn.scheme,
+            self._conn.socket,
+            self.workspace
+        )
+        response = _utils.make_request("POST", url, self._conn, json={"run_id": run.id})
+        _utils.raise_for_http_error(response)
+        build_id = response.json()["id"]
+
+        # Update stages with new build
+        url = "{}://{}/api/v1/deployment/workspace/{}/endpoints/{}/stages/{}/update".format(
+            self._conn.scheme,
+            self._conn.socket,
+            self.workspace,
+            self.id,
+            stage_id
+        )
+        response = _utils.make_request("PUT", url, self._conn, json={"build_id": build_id, 'strategy': strategy._STRATEGY})
+        _utils.raise_for_http_error(response)
+
+    def _get_active_stage(self):
```

Comment:
This seems appropriate 😄 
```suggestion
    def _get_or_create_active_stage(self):
```

Reply:
Fixed.